pkg/asset/rhcos/bootstrap_image.go

```diff
@@ -37,6 +39,13 @@ func (i *BootstrapImage) Generate(p asset.Parents) error {
 	p.Get(ic)
 	config := ic.Config
 
+	// Check for RHCOS image URL override
+	if boi := config.Platform.BareMetal.BootstrapMachineOSImage; boi != "" {
```

Comment:
I would not read from `Platform.BareMetal` unless you're in the case below where we know config.Platform.Name() is baremetal

Reply:
(The reason being Platform is a pointer and BareMetal is null in every other case)

Reply:
Thats correct. I'll add a check before we use the option.